nft/ethereum_payment_obligation.go

```diff
@@ -247,6 +257,25 @@ func (s *ethereumPaymentObligation) minter(ctx context.Context, tokenID TokenID,
 	}
 }
 
+func isNFTMinted(nfts []*coredocumentpb.NFT, registry []byte) bool {
+	for _, nft := range nfts {
+		if bytes.Equal(nft.RegistryId[:20], registry) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func addNFT(cd *coredocumentpb.CoreDocument, registry, tokenID []byte) {
```

Comment:
Maybe move the function to the coredocument package?

Reply:
I wouldn't move anything in to the coredoc package as it is going away soon. This code will be refactored with Charly's work to use the cd model instead.

Reply:
We are getting rid of coredocument package. To much dependency on a package for simple function might not be needed

Reply:
But once we migrate to coreDocModel, we can make all the mutating functions as part of that model 👍